### DISCUSSION TITLE
Partion Wizard not found on Shelly Pro 2 with Tasmota 13.1.0

### DISCUSSION THREAD

**Discussion by RaidObServer** [arendst/Tasmota#19553] [Q&A]
Inital my Shelly Pro 2 device was flashed with Tasmota 12.5.0. After the OTA update to 13.1.0 (without any error) the Partition Wizard is no longer accessible. The button is there and after clicking it the following error message appears:

File Not Found

URI: /part_wiz
Method: GET
Arguments: 0

and also the Configuration Matter is not available:

File Not Found

URI: /matterc
Method: GET
Arguments: 0

Has anyone had such a problem and can help me?

best regards & thanks
Stefan

**Comment by s-hadinger** [comment]
I have never seen this. Please set `Weblog 3` and paste the full log of the boot of your device.

**Reply by RaidObServer** [reply]
00:00:26.234 BRY: Did save calibration data to 'aux.bin' and cloud key data to 'shelly.bin'
00:00:26.252 BRY: No 'autoexec.be'
00:00:26.317 RTC: UTC 2023-09-20T15:04:17, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
16:04:17.000 RTC: Synced by NTP
16:04:17.477 HTP: Web server active on tasmota-8DD0B4-4276-eth with IP address 192.168.1.81
16:04:17.792 MQT: Attempting connection...
16:04:17.805 MQT: Connected
16:04:17.808 MQT: tele/tasmota_8DD0B4/LWT = Online (retained)
16:04:17.811 MQT: cmnd/tasmota_8DD0B4/POWER = 
16:04:17.813 MQT: Subscribe to cmnd/tasmota_8DD0B4/#
16:04:17.816 MQT: Subscribe to cmnd/tasmotas/#
16:04:17.817 MQT: Subscribe to cmnd/DVES_8DD0B4_fb/#
16:04:17.822 MQT: tele/tasmota_8DD0B4/INFO1 = {"Info1":{"Module":"Shelly Pro 2PM","Version":"13.1.0(tasmota32)","FallbackTopic":"cmnd/DVES_8DD0B4_fb/","GroupTopic":"cmnd/tasmotas/"}}
16:04:17.838 MQT: tele/tasmota_8DD0B4/INFO2 = {"Info2":{"WebServerMode":"Admin","Ethernet":{"Hostname":"tasmota-8DD0B4-4276-eth","IPAddress":"192.168.1.81"}}}
16:04:17.856 MQT: tele/tasmota_8DD0B4/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":17}}
16:04:17.861 MQT: stat/tasmota_8DD0B4/RESULT = {"POWER1":"OFF"}
16:04:17.863 MQT: stat/tasmota_8DD0B4/POWER1 = OFF
16:04:17.877 MQT: stat/tasmota_8DD0B4/RESULT = {"POWER2":"OFF"}
16:04:17.879 MQT: stat/tasmota_8DD0B4/POWER2 = OFF
16:04:17.888 TFS: File 'autoexec.bat' not found
16:04:18.900 SRC: MQTT
16:04:18.902 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 1, Data '1'
16:04:18.907 MQT: stat/tasmota_8DD0B4/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:02","StartupUTC":"2023-09-20T15:04:16","Sleep":50,"CfgHolder":4617,"BootCount":16,"BCResetTime":"2023-05-16T09:01:58","SaveCount":251}}
16:04:18.957 SRC: MQTT
16:04:18.959 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 11, Data '11'
16:04:18.968 MQT: stat/tasmota_8DD0B4/STATUS11 = {"StatusSTS":{"Time":"2023-09-20T16:04:18","Uptime":"0T00:00:02","UptimeSec":2,"Heap":102,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":197,"MqttCount":1,"Berry":{"HeapUsed":21,"Objects":289},"POWER1":"OFF","POWER2":"OFF"}}
16:04:19.007 SRC: MQTT
16:04:19.009 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 10, Data '10'
16:04:19.023 MQT: stat/tasmota_8DD0B4/STATUS10 = {"StatusSNS":{"Time":"2023-09-20T16:04:19","Switch1":"OFF","Switch2":"OFF","ANALOG":{"Temperature1":49.1,"Temperature2":54.7},"ENERGY":{"TotalStartTime":"2023-08-13T15:41:52","Total":0.039,"Yesterday":0.000,"Today":0.000,"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":[0,0],"Current":[0.000,0.000]},"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0},"TempUnit":"C"}}
16:04:22.291 MQT: tele/tasmota_8DD0B4/STATE = {"Time":"2023-09-20T16:04:22","Uptime":"0T00:00:06","UptimeSec":6,"Heap":112,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"Berry":{"HeapUsed":21,"Objects":289},"POWER1":"OFF","POWER2":"OFF"}
16:04:22.328 MQT: tele/tasmota_8DD0B4/SENSOR = {"Time":"2023-09-20T16:04:22","Switch1":"OFF","Switch2":"OFF","ANALOG":{"Temperature1":49.2,"Temperature2":54.7},"ENERGY":{"TotalStartTime":"2023-08-13T15:41:52","Total":0.039,"Yesterday":0.000,"Today":0.000,"Period":[0,0],"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":[0,0],"Current":[0.000,0.000]},"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0},"TempUnit":"C"}
16:04:23.303 QPC: Reset
16:04:24.308 NRG: Energy reset by invalid data
16:04:25.296 APP: Boot Count 17
16:04:26.544 CFG: Saved, Count 252, Bytes 4096
16:04:27.239 HTP: Console

**Reply by RaidObServer** [reply]
This is all that is in the console after booting. I have activated Weblog 3 in advance.

**Reply by s-hadinger** [reply]
Sorry but it's missing the important parts, the boot process. Please try to get logs earlier

**Reply by RaidObServer** [reply]
00:00:00.003 HDW: ESP32-D0WDQ6 
00:00:00.058 UFS: FlashFS mounted with 396 kB free
00:00:00.080 CFG: Loaded from File, Count 266
00:00:00.090 QPC: Count 1
00:00:00.092 CFG: CR 512/699, Busy 0
00:00:00.102 SPI: Hardware using GPIO14(CLK), GPIO13(MOSI) and GPIO12(MISO)
00:00:00.105 CFG: CR 512/699, Busy 0
00:00:00.106 CFG: CR 512/699, Busy 0
00:00:00.112 TFS: File 'mcp23x.dat' not found
00:00:00.112 MCP: Invalid template
00:00:00.120 ROT: Mode 1
00:00:00.202 BRY: Berry initialized, RAM used=4502 bytes
00:00:00.256 BRY: No 'preinit.be'
00:00:00.280 CFG: Energy loaded from file
00:00:00.382 SPI: ADE7953 found
00:00:00.388 TFS: File 'calib.dat' not found
00:00:00.388 NRG: Init driver 7
00:00:00.389 Shutter: Use defaults
00:00:00.390 SHT: About to load settings from file /.drvset027
00:00:00.426 SHT: ShutterMode: 1
00:00:00.427 SHT: Shtr1 min realpos_chg: 2124, min tilt_chg 2
00:00:00.428 SHT: Shtr1 Init. Pos 0, Inv 0, Locked 0, Endstop enab 0, webButt inv 0, Motordel: 0
00:00:00.430 SRC: Restart
00:00:00.436 Project tasmota - Tasmota Pro2 RS Kind Fenster Version 13.1.0(tasmota32)-2_0_11(2023-08-15T08:00:54)
00:00:01.772 ETH: Attempting connection...
00:00:01.774 ETH: Connected at 100Mbps Full Duplex, Mac EC:62:60:8B:3C:33, Hostname tasmota-8B3C30-7216-eth
00:00:01.824 HDW: Shelly Pro 2PM initialized
00:00:01.993 TAP: Loaded Tasmota App 'Partition_Wizard.tapp'
00:00:02.525 BRY: No 'autoexec.be'
00:00:02.925 WIF: Attempting connection...
00:00:03.347 ETH: IPv6 Local fe80::ee62:60ff:fe8b:3c33
00:00:03.380 WIF: Connecting to AP1 404 network unavailable Channel 6 BSSId E0:28:6D:75:3F:C2 in mode 11n as tasmota-8B3C30-7216...
00:00:03.445 WIF: Attempting connection...
00:00:04.002 ETH: Disconnected
00:00:04.981 WIF: Attempting connection...
00:00:05.111 WIF: IPv4 192.168.1.151, mask 255.255.255.0, gateway 192.168.1.1
00:00:05.126 ETH: Connected at 100Mbps Full Duplex, Mac EC:62:60:8B:3C:33, Hostname tasmota-8B3C30-7216-eth
00:00:05.130 ETH: IPv4 192.168.1.83, mask 255.255.255.0, gateway 192.168.1.1
00:00:06.446 WIF: Connected
00:00:06.619 RTC: UTC 2023-09-20T19:06:09, DST 2023-03-26T02:00:00, STD 2023-10-29T03:00:00
20:06:09.001 RTC: Synced by NTP
20:06:09.044 HTP: Web server active on tasmota-8B3C30-7216-eth with IP address 192.168.1.83
20:06:10.356 ETH: IPv6 Local fe80::ee62:60ff:fe8b:3c33
20:06:10.647 MQT: Attempting connection...
20:06:10.681 MQT: Connected
20:06:10.687 MQT: tele/tasmota_8B3C30/LWT = Online (retained)
20:06:10.689 MQT: cmnd/tasmota_8B3C30/POWER = 
20:06:10.691 MQT: Subscribe to cmnd/tasmota_8B3C30/#
20:06:10.692 MQT: Subscribe to cmnd/tasmotas/#
20:06:10.694 MQT: Subscribe to cmnd/DVES_8B3C30_fb/#
20:06:10.698 MQT: tele/tasmota_8B3C30/INFO1 = {"Info1":{"Module":"Shelly Pro 2PM","Version":"13.1.0(tasmota32)","FallbackTopic":"cmnd/DVES_8B3C30_fb/","GroupTopic":"cmnd/tasmotas/"}}
20:06:10.714 MQT: tele/tasmota_8B3C30/INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-8B3C30-7216","IPAddress":"192.168.1.151","IP6Global":"","IP6Local":"fe80::ee62:60ff:fe8b:3c30","Ethernet":{"Hostname":"tasmota-8B3C30-7216-eth","IPAddress":"192.168.1.83"}}}
20:06:10.743 MQT: tele/tasmota_8B3C30/INFO3 = {"Info3":{"RestartReason":"Software reset CPU","BootCount":27}}
20:06:10.749 MQT: stat/tasmota_8B3C30/RESULT = {"POWER1":"OFF"}
20:06:10.751 MQT: stat/tasmota_8B3C30/POWER1 = OFF
20:06:10.764 MQT: stat/tasmota_8B3C30/RESULT = {"POWER2":"OFF"}
20:06:10.767 MQT: stat/tasmota_8B3C30/POWER2 = OFF
20:06:10.773 TFS: File 'autoexec.bat' not found
20:06:11.347 WIF: IPv6 Local fe80::ee62:60ff:fe8b:3c30
20:06:11.839 SRC: MQTT
20:06:11.841 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 1, Pld 1, Data '1'
20:06:11.848 MQT: stat/tasmota_8B3C30/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"0T00:00:07","StartupUTC":"2023-09-20T19:06:04","Sleep":50,"CfgHolder":4617,"BootCount":26,"BCResetTime":"2023-05-16T12:47:00","SaveCount":266}}
20:06:11.895 SRC: MQTT
20:06:11.897 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 11, Data '11'
20:06:11.908 MQT: stat/tasmota_8B3C30/STATUS11 = {"StatusSTS":{"Time":"2023-09-20T20:06:11","Uptime":"0T00:00:07","UptimeSec":7,"Heap":77,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":25,"MqttCount":1,"Berry":{"HeapUsed":23,"Objects":302},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"404 network unavailable","BSSId":"E0:28:6D:75:3F:C2","Channel":6,"Mode":"11n","RSSI":52,"Signal":-74,"LinkCount":1,"Downtime":"0T00:00:04"}}}
20:06:11.959 SRC: MQTT
20:06:11.961 CMD: Grp 0, Cmd 'STATUS', Idx 1, Len 2, Pld 10, Data '10'
20:06:11.976 MQT: stat/tasmota_8B3C30/STATUS10 = {"StatusSNS":{"Time":"2023-09-20T20:06:11","Switch1":"OFF","Switch2":"OFF","ANALOG":{"Temperature1":57.2,"Temperature2":66.7},"ENERGY":{"TotalStartTime":"2023-08-13T15:43:42","Total":0.042,"Yesterday":0.000,"Today":0.000,"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":[0,0],"Current":[0.000,0.000]},"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0},"TempUnit":"C"}}
20:06:12.571 QPC: Reset
20:06:13.558 NRG: Energy reset by invalid data
20:06:14.549 APP: Boot Count 27
20:06:15.403 CFG: Saved, Count 267, Bytes 4096
20:06:15.583 MQT: tele/tasmota_8B3C30/STATE = {"Time":"2023-09-20T20:06:15","Uptime":"0T00:00:11","UptimeSec":11,"Heap":78,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":41,"MqttCount":1,"Berry":{"HeapUsed":23,"Objects":302},"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"404 network unavailable","BSSId":"E0:28:6D:75:3F:C2","Channel":6,"Mode":"11n","RSSI":50,"Signal":-75,"LinkCount":1,"Downtime":"0T00:00:04"}}
20:06:15.635 MQT: tele/tasmota_8B3C30/SENSOR = {"Time":"2023-09-20T20:06:15","Switch1":"OFF","Switch2":"OFF","ANALOG":{"Temperature1":57.1,"Temperature2":66.5},"ENERGY":{"TotalStartTime":"2023-08-13T15:43:42","Total":0.042,"Yesterday":0.000,"Today":0.000,"Period":[0,0],"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Voltage":[0,0],"Current":[0.000,0.000]},"Shutter1":{"Position":0,"Direction":0,"Target":0,"Tilt":0},"TempUnit":"C"}

**Reply by RaidObServer** [reply]
sorry my mistake. I hope its better now.

**Comment by Jason2866** [comment]
Delete all files in Tasmota Filesystem. Reset the device (`reset 5`) Do an OTA upgrade again to the latest development version. 
Does it work now?

**Reply by RaidObServer** [reply]
i tried it and it worked until i started the Auto Configuration. Then the problem was again as described.

I have 4 devices with the same problem. Can this be due to the Safeboot?

**Reply by s-hadinger** [reply]
What auto configuration template are you using?

**Reply by RaidObServer** [reply]
Auto Configuration Template: Shelly Pro 2PM

**Reply by s-hadinger** [reply]
I tried to reproduce but I just bricked my device. I don't have a Shelly device

**Reply by RaidObServer** [reply]
can this possibly be due to the Safeboot? For the devices I have updated to Tasmota 13.1.0 without migrating the partitions in the partition wizard. There was also no error message that something does not fit.

**Reply by RaidObServer** [reply]
on your bricked Shelly Pro 2 device Power, Out 1 and Out 2 are lighting red?
I had that before and I could flash it via USB2Serial. It is important that only the board with the ESP32 is alone when flashing. All others must be unplugged before. For this the device must be opened. Then I could flash Tasmota 12.5.0.

**Reply by s-hadinger** [reply]
I don't have any Shelly device, I tried to apply to another device, but I broke the bootloader

**Comment by Jason2866** [comment]
Can reproduce the issue. Investigating.

**Reply by Jason2866** [reply]
Solution: Update the Shelly device to latest firmware (currently v1.0.3) **before** starting the convert process.
Do NOT try anymore to do the conversion via an URL. Use the firmware "drag&drop" update Area in Shelly firmware.
With this workflow we ended in successful converted devices using the safeboot scheme

**Reply by RaidObServer** [reply]
Thanks for your fast Help.
Where can i get the original Shelly 1.0.3 firmware?

**Reply by Jason2866** [reply]
Use the running Shelly firmware to update

**Reply by Jason2866** [reply]
If the device is bricked, just erase the flash and flash Tasmota via serial

**Reply by RaidObServer** [reply]
Hmm, on the Shelly are Tasmota running. Ist there a way to get back to Shelly firmware?

**Reply by RaidObServer** [reply]
Ok, the i will flash Tasmota via Serial and Start direct after Tasmota ist running the convert process.

**Comment by Jason2866** [comment]
> Hmm, on the Shelly are Tasmota running. Ist there a way to get back to Shelly firmware?

No

**Reply by RaidObServer** [reply]
1. shelly flashed via USB2Serial with 12.5.0
2. auto configuration: Shelly Pro 2PM
3. partition wizard: Safeboot Conversion

Shelly uses the Safeboot layout and it works.
That's exactly what I'm doing with the other 3 as well. 
Thanks for your help :-)

**Reply by RaidObServer** [reply]
too early rejoiced. Even with Safeboot layout the same error is present after configuration and reboot.

**Reply by RaidObServer** [reply]
I have another Shelly Pro 2 that I have flashed using the procedure described here (https://github.com/tasmota/mgos32-to-tasmota32) and this also has the problem.
